src/main/scala/ai/privado/languageEngine/javascript/passes/read/GraphqlQueryParserPass.scala

```diff
@@ -0,0 +1,208 @@
+      case null => Nil
+    }
+
+    processGQLReadNode(builder, node, typeName, fields.toList)
+  }
+
+  // Function to extract fields
+  def extractFields(selections: List[Selection]): List[String] = {
+    selections.flatMap {
+      case field: Field =>
+        val nestedFields = extractFields(field.selections.toList)
+        // field.name :: nestedFields.map(nestedField => s"${field.name}.$nestedField")
+        field.name :: nestedFields.map(nestedField => nestedField)
+      case _ => Nil
+    }
+  }
+
+  // Function to extract variable definitions
+  def extractVariableDefinitions(variableDefinitions: List[VariableDefinition]): List[String] = {
+    variableDefinitions.map(definition => definition.name)
+  }
+
+  // Function to extract object type definitions
+  def extractObjectTypeDefinitions(definitions: List[ObjectTypeDefinition]): List[String] = {
+    definitions.map(definition => definition.name)
+  }
+
+  // Function to extract input value definitions
+  def extractInputValueDefinitions(definitions: List[InputValueDefinition]): List[String] = {
+    definitions.map(definition => definition.name)
+  }
+
+  def processGQLReadNode(builder: DiffGraphBuilder, node: Expression, tableName: String, columns: List[String]) = {
+    // Match classes which end with tableName/ObjectName
+    val sensitiveMemberRuleIds = sensitiveClasses.find(s => s.matches(s"(?i).*${tableName}")) match {
+      case Some(value) => sensitiveClassesWithMatchedRules(value).keys.l
+      case None        => List.empty
+    }
+
+    if (columns.length == 1 && columns(0) == "*") {
+      if (sensitiveMemberRuleIds.nonEmpty)
+        sensitiveMemberRuleIds.foreach(ruleId => addTagsToNode(ruleId, node, builder))
+      else {
+        /* Run dataflow and verify the data-elements read from the call,
+              Ex - resultSet = statement.executeQuery("SELECT * FROM mytable");
+              // Loop through the result set and print out each row
+              while (resultSet.next()) {
+                  int id = resultSet.getInt("id");
+                  String firstName = resultSet.getString("name");
+                  int age = resultSet.getInt("age");
+                  System.out.println("ID: " + id + ", Name: " + firstName + ", Age: " + age)
+              }
+         */
+        val dataElementSinks =
+          Dataflow
+            .getSources(cpg)
+            .filterNot(_.isMember)
+            .map(_.asInstanceOf[CfgNode])
+            .l
+
+        val readFlow = dataElementSinks.reachableByFlows(node)(Utilities.getEngineContext(4)).l
+        if (readFlow.nonEmpty) {
+          // As a flow is present from Select query to a Data element we can say, the data element is read from the query
+          readFlow
+            .flatMap(_.elements.last.tag.value("Data.Sensitive.*"))
+            .value
+            .foreach(ruleId => addTagsToNode(ruleId, node, builder))
+        }
+      }
+    } else {
+      if (sensitiveMemberRuleIds.nonEmpty)
```

Comment:
@khemrajrathore Can we just keep the logic inside this `else` group ?
Above one is related to sql queries & not used with GQL.